### DISCUSSION TITLE
[Question] pandas data frame in android apps

### DISCUSSION THREAD

**Discussion by listpplo** [flet-dev/flet#2324] [Q&A] [answered]
### Question

Can any one help me I want to make an app that stores data in a CVS using pandas data-frame in android  and also whenever i build the app it always gives the initial program on Linux i.e it runs the hello, flet !

command : flet built linux

Thanks in Advance 

### Code sample

```python
import flet as ft
import pandas as pd


def main(page: ft.Page):
    try:
        df = pd.read_csv(
            "assets/machine-readable-business-employment-data-sep-2023-quarter.csv"
        )
        df = df[["Series_reference", "Period", "Data_value"]]
        columnHeader = list(df.columns.values)
        df = df.head(20)
        print(df)

    except FileNotFoundError as e:
        print(e)
        columnHeader = ["NotFound"]

    page.scroll = True

    page.add(
        ft.Row(
            [
                ft.DataTable(
                    columns=[ft.DataColumn(ft.Text(header)) for header in columnHeader],
                    rows=[
                        ft.DataRow(
                            cells=[
                                ft.DataCell(ft.Text(f"{df['Series_reference'][ind]}")),
                                ft.DataCell(ft.Text(f"{df['Period'][ind]}")),
                                ft.DataCell(ft.Text(f"{df['Data_value'][ind]}")),
                            ]
                        )
                        for ind in df.index
                    ],
                )
            ],
            scroll=True,
        )
    )


ft.app(main)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by fsmosca** [comment]
You have posted a code, did you attempt to build an apk on it using the command "flet build apk"?

**Reply by listpplo** [reply]
Yes I have build the apk when I tried on my phone it just gives a black screen and also, Is the asset directory is also clubbed to together with the formed package?
Or I need to pack it with some other method ?

**Reply by FeodorFitsner** [reply, marked as answer]
`pandas` is the "native" package and the entire process is not yet automated for Android and iOS. You need to follow this guide to add `pandas` to your Android app: https://flet.dev/docs/guides/python/packaging-app-for-distribution#native-python-packages-for-android-and-ios

**Reply by listpplo** [reply]
Thanks for the reply, much appreciated !! Can you please also tell me how i can access the underlying filesystems , for example  if I need to make a document to save different parameters , how i can do so ?
Like in python can i use os module to get the current location and perform the file operations i.e. file read or write.

**Reply by FeodorFitsner** [reply]
Take a look at this discussion: https://github.com/flet-dev/flet/issues/2189#issuecomment-1839826040
I also created [this issue](https://github.com/flet-dev/flet/issues/2357) to have an "official" implementation.
